File: 14. Longest Common Prefix.cpp
#include<iostream>

using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) {
            return "";
        }

        string first = strs[0];
        string common = "";

        for (int i = 0; i < first.length(); ++i) {
            int commonCount = 0;
            for(const string &str : strs) {
                if (first[i] == str[i]) {
                    commonCount++;
                }
                else {
                    return common;
                }
                if (commonCount == strs.size()) {
                    common += first[i];
                }
            }
        }
        return common;
    }
};

File: 14. Longest Common Prefix.py
class Solution:
    def getMaxLengthStr(strs: list[str]) -> str:
        max = strs[0]
        for i in range(1, len(strs)):
            if len(max) < len(strs[i]):
                max = strs[i]
        return max

    def longestCommonPrefix(self, strs: list[str]) -> str:
        longestStr = Solution.getMaxLengthStr(strs)
        common = ""
        for char in longestStr:
            common_count = 0
            for string in strs:
                if char in string:
                    common_count += 1
                if common_count == len(strs):
                    common += char
            # print(char, common_count, common)

        return common


result = Solution().longestCommonPrefix(["flower", "flow", "flight"])
print(result)
